lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,354 @@
+    necessary to the analysis e.g. A_eff, E_disp and so on...
+        http://gamma-astro-data-formats.readthedocs.io/en/latest/
+
+    This function will create the necessary data format, starting from the path that contains the DL3
+    converted fits file.
+
+    Parameters
+    ----------
+    filename_list : list
+        list of filenames
+    fits_dir : str
+        directory containing the fits file
+    """
+
+    hdu_tables = []
+    obs_tables = []
+
+    #loop through the files
+    for file in filename_list:
+        filepath = fits_dir/file
+        if filepath.is_file():
+            try:
+                event_table = Table.read(filepath, hdu="EVENTS")
+                gti_table = Table.read(filepath, hdu="GTI")
+                pointing_table = Table.read(filepath, hdu="POINTING")
+            except Exception:
+                log.error(f"fits corrupted for file {file}")
+                continue
+        else:
+            log.error(f"fits {file} doesn't exist")
+            continue
+
+        #The column names for the table follows the scheme as shown in
+        #https://gamma-astro-data-formats.readthedocs.io/en/latest/general/HDU_CLASS.html
+        ###############################################
+        # Event list
+        t_events = Table(
+                {
+                'OBS_ID':[event_table.meta['OBS_ID']],
+                'HDU_TYPE':['events'],
+                'HDU_CLASS':['events'],
+                'HDU_CLASS2':[''],
+                'HDU_CLASS3':[''],
+                'HDU_CLASS4':[''],
+                'FILE_DIR':[''],
+                'FILE_NAME': [file],
+                'HDU_NAME':['events']},
+                dtype=('>i8', 'S6', 'S10',
+                'S20', 'S20', 'S20', 'S70',
+                'S54', 'S20')
+                )
+        hdu_tables.append(t_events)
+        ###############################################
+        #GTI
+        t_gti = t_events.copy()
+
+        t_gti['HDU_TYPE'] = 'gti'
+        t_gti['HDU_CLASS'] = 'gti'
+        t_gti['HDU_NAME'] = 'gti'
+
+        hdu_tables.append(t_gti)
+        ###############################################
+        #POINTING
+        t_pnt = t_events.copy()
+
+        t_pnt['HDU_TYPE'] = ['pointing']
+        t_pnt['HDU_CLASS'] = ['pointing']
+        t_pnt['HDU_NAME'] = ['pointing']
+
+        hdu_tables.append(t_pnt)
+        ###############################################
+        #Energy Dispersion
+        try:
+            Table.read(filepath, hdu="ENERGY DISPERSION")
+            t_edisp = t_events.copy()
+
+            t_edisp['HDU_TYPE'] = ['edisp']
+            t_edisp['HDU_CLASS'] = ['edisp_2d']
+            t_edisp['HDU_CLASS2'] = ['EDISP']
+            t_edisp['HDU_CLASS3'] = ['POINT-LIKE']
+            t_edisp['HDU_CLASS4'] = ['EDISP_2D']
+            t_edisp['HDU_NAME'] = ['ENERGY DISPERSION']
+
+            hdu_tables.append(t_edisp)
+        except:
```

Comment:
Those excepts are very general and the error message might not always be appropriate

Reply:
Agreed. I have updated them. Let me know if it could be further improved